document-translator/script.js

```diff
@@ -0,0 +1,155 @@
+  img.src = '';
+
+  let file;
+  if (!useExample) {
+    try {
+      const [handle] = await showOpenFilePicker({
+        types: [
+          {
+            description: 'Image or PDF files',
+            accept: {
+              'image/*': ['.png', '.gif', '.jpeg', '.jpg', '.webp', '.avif'],
+              'application/pdf': ['.pdf'],
+            },
+          },
+        ],
+      });
+      file = await handle.getFile();
+      if (file.type === 'application/pdf') {
+        const pdfBytes = new Uint8Array(await file.arrayBuffer());
+        file = await convertFirstPageToPngBlob(pdfBytes);
+      }
+    } catch (err) {
+      console.error(err.name, err.message);
+    }
+  } else {
+    useExample = false;
+    file = await fetch('specimen.png').then(response => response.blob());
+  }
+  const blobURL = URL.createObjectURL(file);
+  img.src = blobURL;
+  setTimeout(() => {
+    URL.revokeObjectURL(blobURL);
+  }, 0);
+
+  try {
+    const session = await LanguageModel.create({
+      expectedInputs: [{ type: 'image' }, { type: 'text' }],
+    });
+    const stream = session.promptStreaming([
+      {
+        role: 'user',
+        content: [
+          {
+            type: 'text',
+            value: 'Extract all text from the provided image.',
+          },
+          { type: 'image', value: file },
+        ],
+      },
+    ]);
+    for await (const chunk of stream) {
+      output.append(chunk);
+    }
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+});
+
+languageButton.addEventListener('click', async () => {
+  if (!output.innerText.length) {
+    return;
+  }
+
+  detectedLanguage = false;
+  detectedLanguageParagraph.innerHTML = '';
+
+  try {
+    const languageDetector = await LanguageDetector.create();
+    ({ detectedLanguage } = (
+      await languageDetector.detect(output.innerText)
+    )[0]);
+    const displayLanguage = new Intl.DisplayNames(['en'], {
+      type: 'language',
+    }).of(detectedLanguage);
+    detectedLanguageParagraph.textContent = displayLanguage;
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+});
+
+translateButton.addEventListener('click', async () => {
+  if (!detectedLanguage || detectedLanguage === 'en') {
+    return;
+  }
+
+  translationParagraph.innerHTML = '';
+
+  try {
+    const translator = await Translator.create({
+      sourceLanguage: detectedLanguage,
+      targetLanguage: 'en',
+    });
+    const paragraphs = output.innerText.split('\n');
+    for (const paragraph of paragraphs) {
+      if (!paragraph) {
+        translationParagraph.append('\n');
+        continue;
+      }
+      const translateStream = translator.translateStreaming(paragraph);
+      for await (const chunk of translateStream) {
+        translationParagraph.append(chunk);
+      }
+      translationParagraph.append('\n');
+    }
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+});
+
+exampleButton.addEventListener('click', async () => {
```

Comment:
`async` is not needed there.

Reply:
You're right, I deleted the `await`, but forgot about the `async`.